src/performance.js

```diff
@@ -226,45 +267,57 @@ export class Performance {
     }
 
     this.events_.push({
-      label: label,
-      opt_from: opt_from,
-      opt_value: opt_value
+      'label': label,
+      'from': opt_from,
+      'value': opt_value,
     });
   }
 
 
-  /** @private */
+  /**
+   * Forwards all queued ticks to the viewer tick method.
+   * @private
+   */
   flushQueuedTicks_() {
-    if (!this.tick_) {
+    if (!this.viewer_) {
       return;
     }
 
     this.events_.forEach(tickEvent => {
-      this.tick_(tickEvent.label, tickEvent.opt_from, tickEvent.opt_value);
+      this.viewer_.tick(tickEvent);
     });
     this.events_.length = 0;
   }
 
 
   /**
-   * Sets the `tick` function.
-   *
-   * @param {funtion(string,?string=,number=)} tick function that the tick
-   *   events get forwarded to. Function can take in a `label` as the first
-   *   argument and an optional `opt_from` label to use
-   *   as a relative start for this tick. A third argument `opt_value` can
-   *   also be provided to indicate when to record the tick at.
-   * @param {function()=} opt_flush callback function that is called
-   *   when we are ready for the ticks to be forwarded to an endpoint.
+   * Calls setParams with relevant document information.
+   * @return {!Promise}
+   * @private
    */
-  setTickFunction(tick, opt_flush) {
-    this.tick_ = tick;
-    this.flush_ = opt_flush;
-    this.flushQueuedTicks_();
-    // We need to call flush right away in case `setTickFunction` is called
-    // later than the amp codebase had invoked the performance services'
-    // `flush` method to forward ticks.
-    this.flush();
+  setDocumentInfoParams_() {
+    return this.whenViewportLayoutComplete_().then(() => {
+      const params = Object.create(null);
+      let sourceUrl = documentInfoFor(this.win).sourceUrl;
+      const hashIndex = sourceUrl.indexOf('#');
+      if (hashIndex > -1) {
```

Comment:
Doing a simple replace would be more concise here: `sourceUrl.replace(/#.*/, '')`